src/applications/vaos/appointment-list/redux/selectors.js

```diff
@@ -374,51 +370,19 @@ export function selectExpressCareAvailability(state) {
   };
 }
 
-export function selectExpressCareRequestById(state, id) {
-  const { appointmentDetails, pending, past, confirmed } = state.appointments;
-
-  if (appointmentDetails[id]) {
-    return appointmentDetails[id];
-  }
-
-  const allAppointments = []
-    .concat(pending)
-    .concat(past)
-    .concat(confirmed)
-    .filter(item => !!item);
-
-  return allAppointments.find(p => p.id === id);
-}
-
-export function selectRequestById(state, id) {
-  const { appointmentDetails, pending } = state.appointments;
-
-  if (
-    appointmentDetails[id] &&
-    (appointmentDetails[id].vaos.appointmentType ===
-      APPOINTMENT_TYPES.ccRequest ||
-      appointmentDetails[id].vaos.appointmentType === APPOINTMENT_TYPES.request)
-  ) {
-    return appointmentDetails[id];
-  }
-
-  return pending?.find(p => p.id === id);
-}
-
-export function selectConfirmedAppointmentById(state, id) {
-  const { appointmentDetails, past, confirmed } = state.appointments;
+export function selectAppointmentById(state, id, types = null) {
+  const { appointmentDetails, past, confirmed, pending } = state.appointments;
 
   if (
     appointmentDetails[id] &&
-    (appointmentDetails[id].vaos.appointmentType ===
-      APPOINTMENT_TYPES.vaAppointment ||
-      appointmentDetails[id].vaos.appointmentType ===
-        APPOINTMENT_TYPES.ccAppointment)
+    (types === null ||
+      types.includes(appointmentDetails[id].vaos.appointmentType))
   ) {
     return appointmentDetails[id];
   }
 
   const allAppointments = []
+    .concat(pending)
     .concat(past)
     .concat(confirmed)
     .filter(item => !!item);
```

Comment:
What does the following line do?
```
.filter(item => !!item)
```

Reply:
It removes any items in the array that are `undefined`. Sometimes using concat, if the item you use is an argument is undefined, you can get an array that looks like `[1, undefined, 2]`. This removes those undefined items.

I think in this case one of those arrays was sometimes undefined and that's why I did this, but I don't remember exactly, since I copied it over from one of the other selectors.